src/zenml/integrations/kubeflow/flavors/kubeflow_orchestrator_flavor.py

```diff
@@ -155,43 +155,106 @@ class KubeflowOrchestratorConfig(  # type: ignore[misc] # https://github.com/pyd
     """Configuration for the Kubeflow orchestrator.
 
     Attributes:
-        kubeflow_pipelines_ui_port: A local port to which the KFP UI will be
-            forwarded.
         kubeflow_hostname: The hostname to use to talk to the Kubeflow Pipelines
             API. If not set, the hostname will be derived from the Kubernetes
             API proxy.
         kubeflow_namespace: The Kubernetes namespace in which Kubeflow
             Pipelines is deployed. Defaults to `kubeflow`.
         kubernetes_context: Optional name of a kubernetes context to run
             pipelines in. If not set, will try to spin up a local K3d cluster.
+        local: If `True`, the orchestrator will assume it is connected to a
+            local kubernetes cluster and will perform additional validations and
+            operations to allow using the orchestrator in combination with other
+            local stack components that store data in the local filesystem
+            (i.e. it will mount the local stores directory into the pipeline
+            containers).
         skip_local_validations: If `True`, the local validations will be
             skipped.
-        skip_cluster_provisioning: If `True`, the k3d cluster provisioning will
-            be skipped.
-        skip_ui_daemon_provisioning: If `True`, provisioning the KFP UI daemon
-            will be skipped.
-        container_registry_name: The name of the container registry stack
-            component to use. If not specified, the container registry
-            in the active stack is used.
     """
 
-    kubeflow_pipelines_ui_port: int = DEFAULT_KFP_UI_PORT
     kubeflow_hostname: Optional[str] = None
     kubeflow_namespace: str = "kubeflow"
-    kubernetes_context: Optional[str] = None  # TODO: Potential setting
+    kubernetes_context: str  # TODO: Potential setting
+    local: bool = False
     skip_local_validations: bool = False
-    skip_cluster_provisioning: bool = False
-    skip_ui_daemon_provisioning: bool = False
-
-    # IMPORTANT: This is a temporary solution to allow the Kubeflow orchestrator
-    # to be provisioned as an individual component (i.e. same as running
-    # `zenml orchestrator kubeflow up`) rather than needing it to be part of the
-    # active stack (i.e. instead of running `zenml stack up`). This is required
-    # by the test framework because the way it works is that it first provisions
-    # the stack components individually, then each test gets a different stack
-    # with the components it requires.
-    # Do not use for anything else! This will be removed in the near future.
-    container_registry_name: Optional[str] = None
+
+    @root_validator(pre=True)
+    def _validate_deprecated_attrs(
```

Comment:
Perhaps `utils.deprecation_utils.deprecate_pydantic_attributes` could be useful here to simplify this validator a bit?